shared_model/interfaces/transaction_responses/tx_response.hpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_TX_RESPONSE_HPP
+#define IROHA_TX_RESPONSE_HPP
+
+#include <boost/variant.hpp>
+#include "interfaces/hashable.hpp"
+#include "interfaces/polymorphic_wrapper.hpp"
+#include "interfaces/transaction_responses/committed_tx_response.hpp"
+#include "interfaces/transaction_responses/stateful_failed_tx_response.hpp"
+#include "interfaces/transaction_responses/stateful_valid_tx_response.hpp"
+#include "interfaces/transaction_responses/stateless_failed_tx_response.hpp"
+#include "interfaces/transaction_responses/stateless_valid_tx_response.hpp"
+#include "interfaces/transaction_responses/unknown_tx_response.hpp"
+#include "interfaces/visitor_apply_for_all.hpp"
+#include "model/transaction_response.hpp"
+
+namespace shared_model {
+  namespace interface {
+    /**
+     * Architecture note: TransactionResponse inherit from hashable, because tx
+     * response should hold hash of transaction for what it attached.
+     */
+    class TransactionResponse
+        : public Hashable<TransactionResponse,
+                          iroha::model::TransactionResponse> {
+     private:
+      /// PolymorphicWrapper shortcut type
+      template <typename Value>
+      using w = detail::PolymorphicWrapper<Value>;
+
+     public:
+      /// Type of variant, that handle all concrete tx responses in the system
+      using ResponseVariantType = boost::variant<w<CommittedTxResponse>,
```

Comment:
It looks like we don't have equivalent statuses for old ```ON_PROCESS``` and ```NOT_RECEIVED```. The last one is important I think.

Reply:
I fixed `NOT_RECEIVED`. `ON_PROCESS` intentionally ignored, with the new shared model design we don't have separated phase for stateless validation.